File: urbanairship-core/src/main/java/com/urbanairship/modules/preferencecenter/PreferenceCenterModuleFactory.java
package com.urbanairship.modules.preferencecenter;

import android.content.Context;

import com.urbanairship.AirshipVersionInfo;
import com.urbanairship.PreferenceDataStore;
import com.urbanairship.PrivacyManager;
import com.urbanairship.modules.Module;
import com.urbanairship.remotedata.RemoteData;

import androidx.annotation.NonNull;
import androidx.annotation.RestrictTo;

/**
 * PreferenceCenter module factory.
 *
 * @hide
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
public interface PreferenceCenterModuleFactory extends AirshipVersionInfo {

    @NonNull
    Module build(
            @NonNull Context Context,
            @NonNull PreferenceDataStore dataStore,
            @NonNull PrivacyManager privacyManager,
            @NonNull RemoteData remoteData
    );
}

File: urbanairship-core/src/main/java/com/urbanairship/modules/preferencecenter/PreferenceCenterModuleFactory.kt
package com.urbanairship.modules.preferencecenter

import android.content.Context
import androidx.annotation.RestrictTo
import com.urbanairship.AirshipVersionInfo
import com.urbanairship.PreferenceDataStore
import com.urbanairship.PrivacyManager
import com.urbanairship.inputvalidation.AirshipInputValidation
import com.urbanairship.modules.Module
import com.urbanairship.remotedata.RemoteData

/**
 * PreferenceCenter module factory.
 *
 * @hide
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
public interface PreferenceCenterModuleFactory : AirshipVersionInfo {

    public fun build(
        context: Context,
        dataStore: PreferenceDataStore,
        privacyManager: PrivacyManager,
        remoteData: RemoteData,
        inputValidator: AirshipInputValidation.Validator
    ): Module
}
